ledger/participant-integration-api/src/main/scala/platform/store/appendonlydao/events/BufferedTransactionsReader.scala

```diff
@@ -0,0 +1,211 @@
+      toApiTx = toFlatTransaction,
+      apiResponseCtor = GetTransactionsResponse(_),
+      fetchTransactions = delegate.getFlatTransactions(_, _, _, _)(loggingContext),
+      sourceTimer = metrics.daml.services.index.getFlatTransactionsSource,
+      resolvedFromBufferCounter =
+        metrics.daml.services.index.streamsBuffer.flatTransactionEventsResolvedFromBuffer,
+      totalRetrievedCounter =
+        metrics.daml.services.index.streamsBuffer.totalFlatTransactionsRetrieved,
+      bufferSizeCounter =
+        // TODO in-memory fan-out: Specialize the metric per consumer
+        metrics.daml.services.index.flatTransactionsBufferSize,
+      outputStreamBufferSize = outputStreamBufferSize,
+    )
+
+  override def getTransactionTrees(
+      startExclusive: Offset,
+      endInclusive: Offset,
+      requestingParties: Set[Party],
+      verbose: Boolean,
+  )(implicit
+      loggingContext: LoggingContext
+  ): Source[(Offset, GetTransactionTreesResponse), NotUsed] =
+    getTransactions(transactionsBuffer)(startExclusive, endInclusive, requestingParties, verbose)(
+      toApiTx = toTransactionTree,
+      apiResponseCtor = GetTransactionTreesResponse(_),
+      fetchTransactions = delegate.getTransactionTrees(_, _, _, _)(loggingContext),
+      sourceTimer = metrics.daml.services.index.getTransactionTreesSource,
+      resolvedFromBufferCounter =
+        metrics.daml.services.index.streamsBuffer.transactionTreeEventsResolvedFromBuffer,
+      totalRetrievedCounter =
+        metrics.daml.services.index.streamsBuffer.totalTransactionTreesRetrieved,
+      bufferSizeCounter =
+        // TODO in-memory fan-out: Specialize the metric per consumer
+        metrics.daml.services.index.transactionTreesBufferSize,
+      outputStreamBufferSize = outputStreamBufferSize,
+    )
+
+  override def lookupFlatTransactionById(
+      transactionId: TransactionId,
+      requestingParties: Set[Party],
+  )(implicit loggingContext: LoggingContext): Future[Option[GetFlatTransactionResponse]] =
+    delegate.lookupFlatTransactionById(transactionId, requestingParties)
+
+  override def lookupTransactionTreeById(
+      transactionId: TransactionId,
+      requestingParties: Set[Party],
+  )(implicit loggingContext: LoggingContext): Future[Option[GetTransactionResponse]] =
+    delegate.lookupTransactionTreeById(transactionId, requestingParties)
+
+  override def getActiveContracts(activeAt: Offset, filter: FilterRelation, verbose: Boolean)(
+      implicit loggingContext: LoggingContext
+  ): Source[GetActiveContractsResponse, NotUsed] =
+    delegate.getActiveContracts(activeAt, filter, verbose)
+
+  override def getContractStateEvents(startExclusive: (Offset, Long), endInclusive: (Offset, Long))(
+      implicit loggingContext: LoggingContext
+  ): Source[((Offset, Long), ContractStateEvent), NotUsed] =
+    throw new UnsupportedOperationException(
+      s"getContractStateEvents is not supported on ${getClass.getSimpleName}"
```

Comment:
Shouldn't this be delegated as well?

Reply:
No, since this stream has no use when in-memory fan-out is enabled. It's use would reveal a bug